packages/engine/src/world/SketchManager.ts

```diff
@@ -35,14 +35,21 @@ export class SketchManager {
   }
 
   public removeSketchFromScene = (instanceId: string): void => {
-    const scene = getDebugScene().scene
-    const oldSketch = scene.getObjectByName(instanceId)
+    const engineScene = getDebugScene()
+    const threeScene = engineScene.scene
+    const oldSketch = threeScene.getObjectByName(instanceId)
 
     if (!oldSketch) {
+      const pass = engineScene.getPassesByName(instanceId)
+      if (pass?.length) {
+        pass.forEach((p) => engineScene.removePass(p))
+        delete this.sketchInstances[instanceId]
+        return
+      }
       throw new Error(`couldn't find sketch to remove: ${instanceId}`)
     }
 
-    scene.remove(oldSketch)
+    threeScene.remove(oldSketch)
```

Comment:
Wondering if we may need to still call `threeScene.remove` sometimes even when there are passes on the sketch?

Is it not quite likely that some sketches may have both postprocessing and 3D objects? Rather than immediately returning at the end of the passes stuff, I'm wondering if it makes sense to shuffle the logic a bit so the code keeps keeps executing to the end of the method. You could optionally call `threeScene?.remove()`. Also wouldn't need to repeat the `delete this.sketchInstances[instanceId]` as it will just naturally get there anyway.